test/e2e/upgrades/upgrade.go

```diff
@@ -41,6 +41,9 @@ const (
 
 // Test is an interface for upgrade tests.
 type Test interface {
+	// Name should return a test name sans spaces.
+	Name() string
```

Comment:
Just to clarify - this is what I was proposing:

[test/e2e/cluster_upgrade.go#52](https://github.com/kubernetes/kubernetes/pull/42588/files#diff-f867f24c5173814d3a4122a401752368R52)
```go
name := reflect.TypeOf(t).Elem().Name()  // e.g. name := "JobUpgradeTest"
testFrameworks[name] = framework.NewDefaultFramework(name)
```